### PR TITLE
Default to Deployment.Size 1

### PR DESCRIPTION
**Pull request by ruromero**

### CODE REVIEW

**Comment by gtully** [comment]
I think this is a very sensible default, but there must be some doc impact here? docs/help/...

File: controllers/activemqartemis_reconciler.go [hunk 1]
```diff
@@ -2574,20 +2559,24 @@ func getValidCondition(cr *brokerv1beta1.ActiveMQArtemis) metav1.Condition {
 }
 
 func getDeploymentCondition(cr *brokerv1beta1.ActiveMQArtemis, podStatus olm.DeploymentStatus) metav1.Condition {
-	if cr.Spec.DeploymentPlan.Size == 0 {
+	deploymentSize := DefaultDeploymentSize
+	if cr.Spec.DeploymentPlan.Size != nil {
+		deploymentSize = *cr.Spec.DeploymentPlan.Size
+	}
```
Comment (brusdev) [inline controllers/activemqartemis_reconciler.go:2565]:
```suggestion
	deploymentSize := getDeploymentSize(cr)
```
Could `getDeploymentSize` be used here?

Comment (ruromero) [reply controllers/activemqartemis_reconciler.go]: Indeed, this is a refactor leftover. Thanks for the catch.

File: examples/artemis/artemis_single.yaml [hunk 2]
```diff
@@ -2,7 +2,3 @@ apiVersion: broker.amq.io/v1beta1
 kind: ActiveMQArtemis
 metadata:
   name: artemis-broker
-spec:
-  deploymentPlan:
-    size: 1
```
Comment (brusdev) [inline examples/artemis/artemis_single.yaml:7]: Should the size field be preserved  at least in 1 example?

**Review by brusdev — APPROVED** [review]

File: examples/artemis/artemis_single.yaml (hunk 2 above)
Comment (ruromero) [reply examples/artemis/artemis_single.yaml:7]: added a new example with `size: 0`

**Review by gtully — APPROVED** [review]

**Comment by gtully** [comment]
done a rebase

**Comment by gtully** [comment]
need to investigate a test, seems we are doing extra reconciles, don't know the root cause yet.

**Pull request merged**